guides/source/active_record_callbacks.md

```diff
@@ -329,46 +725,162 @@ Just as with validations, it is also possible to skip callbacks by using the fol
+method every time a new user is created.
+
+To create a user without sending the welcome email, we can use the
+`ActiveRecord::Suppressor` module as follows:
+
+```ruby
+User.suppress do
+  User.create(name: "Jane", email: "jane@example.com")
+end
+```
+
+In the above code, the `User.suppress` block ensures that the
+`send_welcome_email` callback is not executed during the creation of the "Jane"
+user, allowing us to create the user without sending the welcome email.
+
+WARNING: Using the Active Record Suppressor, while potentially beneficial for
+selectively controlling callback execution, can introduce complexity and
+unexpected behavior. Suppressing callbacks can obscure the intended flow of your
+application, leading to difficulties in understanding and maintaining the
+codebase over time. Carefully consider the implications of suppressing
+callbacks, ensuring thorough documentation and thoughtful testing to mitigate
+risks of unintended side effects, performance issues, and test failures.
 
 Halting Execution
 -----------------
 
-As you start registering new callbacks for your models, they will be queued for execution. This queue will include all your model's validations, the registered callbacks, and the database operation to be executed.
+As you start registering new callbacks for your models, they will be queued for
+execution. This queue will include all of your model's validations, the
+registered callbacks, and the database operation to be executed.
 
-The whole callback chain is wrapped in a transaction. If any callback raises an exception, the execution chain gets halted and a ROLLBACK is issued. To intentionally stop a chain use:
+The whole callback chain is wrapped in a transaction. If any callback raises an
+exception, the execution chain gets halted and a **rollback** is issued. To
+intentionally halt a chain use:
 
 ```ruby
 throw :abort
 ```
 
-WARNING. Any exception that is not `ActiveRecord::Rollback` or `ActiveRecord::RecordInvalid` will be re-raised by Rails after the callback chain is halted. Additionally, may break code that does not expect methods like `save` and `update` (which normally try to return `true` or `false`) to raise an exception.
+WARNING. Any exception that is not `ActiveRecord::Rollback` or
+`ActiveRecord::RecordInvalid` will be re-raised by Rails after the callback
+chain is halted. Additionally, it may break code that does not expect methods
+like `save` and `update` (which normally try to return `true` or `false`) to
+raise an exception.
 
-NOTE: If an `ActiveRecord::RecordNotDestroyed` is raised within `after_destroy`, `before_destroy` or `around_destroy` callback, it will not be re-raised and the `destroy` method will return `false`.
+NOTE: If an `ActiveRecord::RecordNotDestroyed` is raised within `after_destroy`,
+`before_destroy` or `around_destroy` callback, it will not be re-raised and the
+`destroy` method will return `false`.
 
 Relational Callbacks
```

Comment:
I see your point, it is a bit unclear! Reading the descriptions of the Callbacks  - I would have thought that the Relational Callback is actually an Association Callback. However, I worry that changing the name may confuse people because they are already referring to this type of callback as a Relational Callback.

Reply:
I don't have a strong opinion on this. Googling for "Relational Callbacks" doesn't return much relevant results (besides copies of the guides): a [Rails PR](https://github.com/rails/rails/issues/37273) that's actually about association callbacks 🙃 and a [Dev.to article](https://dev.to/justalever/understanding-active-record-callbacks-2l2g).
I'm not a fan of "Cascading Callbacks" either, so I'm ok with keeping "Relational Callbacks".

That dev.to article does have a nice simple definition of "Relational Callbacks" though:
"Much like Active Record relations, callbacks can be performed when related objects are changed."
Maybe we can change the next line to something like the following?
"Callbacks can be performed when related objects are changed."
Feel free to ignore it though.

Reply:
This is a difficult one but they are confusing and should probably be addressed, here's what I did and let me know your thoughts: 

I think both sections are talking about 'associations', but we use 'relations' as sort of a synonym for 'associations' which confuses me a little bit into thinking we may be talking about different base concepts. 

I kept the 'Association Callbacks' title as it is, however, I changed 'Relational Callbacks' to 'Cascading Association Callbacks' and moved the section below `Association Callbacks'. I also changed the definition as per your suggestion and changed terms like 'relationship' to 'association'.

What do you think?